### PR TITLE
Add metadata service as a component of Kubeflow

### PR DESCRIPTION
**Pull request by zhenghuiwang**
Fixes https://github.com/kubeflow/metadata/issues/67

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/3590)

### CODE REVIEW

File: testing/kfctl/kf_is_ready_test.py [hunk 1]
```diff
@@ -60,12 +63,12 @@ def test_kf_is_ready(namespace, use_basic_auth, use_istio):
   # TODO(jlewi): Might want to parallelize this.
   for deployment_name in deployment_names:
     logging.info("Verifying that deployment %s started...", deployment_name)
-    util.wait_for_deployment(api_client, namespace, deployment_name)
-  
+    util.wait_for_deployment(api_client, namespace, deployment_name, 10)
```
Comment (lluunn) [inline testing/kfctl/kf_is_ready_test.py]: what does 10 mean here?

Comment (zhenghuiwang) [reply testing/kfctl/kf_is_ready_test.py]: It means timeout in 10mins if the deployment hasn't started yet.

**Pull request merged**